Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp

```diff
@@ -318,26 +322,46 @@ namespace ROS2::Utils
             }
 
             AZ::IO::Path resolvedPath(resolved);
+            const bool needsVisual = visuals.contains(unresolvedUrfFileName);
             const bool needsCollider = colliders.contains(unresolvedUrfFileName);
 
             const AZStd::string pathSuffix = needsCollider ? "_collider" : "";
             const AZStd::string newFilename = resolvedPath.Stem().String() + pathSuffix + resolvedPath.Extension().String();
             AZ::IO::Path targetPathAssetDst(importDirectoryDst / newFilename);
+            AZ::IO::Path targetPathAssetTmp(importDirectoryTmp / newFilename);
 
             AZ::IO::Path targetPathAssetInfo(targetPathAssetDst.Native() + ".assetinfo");
 
             if (!fileIO->Exists(targetPathAssetDst.c_str()))
             {
-                const auto outcomeMoveDst = fileIO->Copy(resolvedPath.c_str(), targetPathAssetDst.c_str());
+                const auto outcomeCopyTmp = fileIO->Copy(resolvedPath.c_str(), targetPathAssetTmp.c_str());
                 AZ_Printf(
                     "CopyAssetForURDF",
                     "Copy %s to %s, result: %d",
                     resolvedPath.c_str(),
-                    targetPathAssetDst.c_str(),
-                    outcomeMoveDst.GetResultCode());
-                if (outcomeMoveDst)
+                    targetPathAssetTmp.c_str(),
+                    outcomeCopyTmp.GetResultCode());
+                if (outcomeCopyTmp)
                 {
-                    copiedFiles[unresolvedUrfFileName] = targetPathAssetDst.String();
+
+                    // create asset info at destination
+                    const bool assetInfoOk =
+                        CreateSceneManifest(targetPathAssetTmp.String(), targetPathAssetInfo.String(), needsCollider, needsVisual);
+
+                    if (assetInfoOk)
+                    {
+                        const auto outcomeMoveDst = fileIO->Copy(targetPathAssetTmp.c_str(), targetPathAssetDst.c_str());
+                        AZ_Printf(
+                            "CopyAssetForURDF",
+                            "Move %s to %s, result: %d",
+                            targetPathAssetTmp.c_str(),
+                            targetPathAssetDst.c_str(),
+                            outcomeMoveDst.GetResultCode());
+                        if (outcomeMoveDst)
+                        {
+                            copiedFiles[unresolvedUrfFileName] = targetPathAssetDst.String();
+                        }
+                    };
```

Comment:
There might be a race condition here where the AP will start processing the first copied file before the .assetinfo file is copied, which I think would cause an initial export without the .assetinfo to occur before exporting a second time with the settings from the .assetinfo. It might be avoidable by copying the assetinfo first, and the real file second.

Reply:
Thanks for a suggestion @mbalfour-amzn. But I have to disagree. 
Asset info is not copied, it is created in place using  `targetPathAssetInfo`.

So in the moment of creation of assetinfo:
 - the mesh file is still in the temp directory (ignored by AP)
 - `.assetinfo` is created in destination directory.

 Next step is to move (no copy as @lumberyard-employee-dm suggested) mesh file from temp directory to the destination directory.
 AP should pick up the mesh file with `assetinfo` already in place.

Reply:
Of course 
```
CreateSceneManifest(targetPathAssetTmp.String(), targetPathAssetInfo.String(), needsCollider, needsVisual);
``` runs synchronously.

Reply:
I also found some slight possibility for a race in another place. I've added `WaitForAPAndEscalate` function.
There is a call 'GetAvailableAssetInfo' that collects some info about source assets (mainly uids). To make sure that result will be complete, I proposed a function `WaitForAPAndEscalate` that will wait for AP to start report any status on freshly moved source asset.

Reply:
Sorry, I misread the logic here, thanks for the correction. You're absolutely right. 

I've got a suggestion below for how to improve WaitForAPAndEscalate though.